dags/sql/create-function-generic_find_gaps.sql

```diff
@@ -0,0 +1,109 @@
+--noqa: disable=TMP, PRS
+CREATE OR REPLACE FUNCTION public.generic_find_gaps(
+    start_date date,
+    end_date date,
+    id_col text,
+    dt_col text,
+    sch_name text,
+    tbl_name text,
+    gap_threshold interval,
+    default_bin interval,
+    id_col_dtype anyelement DEFAULT NULL::int
```

Comment:
kinda frustrating that it needs to know the datatype at call and we have to add another input to specify the datatype.... I tried some other things but I think this is the only way at the moment.

Reply:
could we do function overloading?

Reply:
I think `id_col_dtype` is more elegant than function overloading!